netket/models/rbm.py

```diff
@@ -195,7 +196,7 @@ def __call__(self, x):
 class RBMSymm(nn.Module):
     """A symmetrized RBM using the :ref:`netket.nn.DenseSymm` layer internally."""
 
-    symmetries: Callable[[], Array]
+    symmetries: HashableArray
```

Comment:
Ideally we should define a type in nk.utils.types named something like `SymmetryLike=Union[SymmetryGroup, HashableArray]` and use that?

Also, if we were to provide a constructor to SymmetryGroup so that one can do 
```python
symm = g.translations()
symm_mat = np.array(symm_mat)
symm_2 = SymmetryGroup(symmetry_mat)
```

we could get rid of the build_RBMSymm method, and just instruct users that if they want to build their symmetries themselves, they must wrap them in a symmetric group

Reply:
what do you think?

Reply:
> Ideally we should define a type in nk.utils.types named something like SymmetryLike=Union[SymmetryGroup, HashableArray] and use that?

That seems like the wrong way around to me. The `SymmGroup` is convertible to an array using `__array__` so it is relatively inconspicuous in an array context. We could have a type `ArrayLike` maybe?

Really, the `HashableArray` -- although only needed in `DenseSymm` etc. right now -- is relatively independent of the symmetries - there are other reasons a model may need an array-valued attribute.

Reply:
yeah, I'm not against HashableArray.
I'm simply arguing that where we need symmetries we can just use them?

Reply:
I do see your point - on the other hand, you can in principle use any array of permutations in `DenseSymm` regardless of whether it arises from a symmetry group or not so (and a `SymmGroup` can always be turned into an array), so using `HashableArray` is more flexible.

Reply:
Ok, we could make `DenseSymm` take either `SymmGroup` and `HashableArray` an get rid of the constructors, though.

The only think we lose is the ability of passing a graph directly. But calling `.automorphisms()` or `.translations()` is not that hard (and it makes it clear which symmetries are used - as we've seen in the 4x4 case, `.automorphisms()` can have surprising results).

Reply:
Indeed.
I don't think that passing a graph directly is a good idea. Better be explicit.
And this makes a simpler UI.

Reply:
This is done now, @PhilipVinc.